elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/parser/ElideConfigParser.java

```diff
@@ -5,99 +5,48 @@
-      e.printStackTrace();
-      log.error(e.getMessage());
-        }
-    }
-
-    private void parse(String localConfigPath) throws Exception {
-        //variables
-        String varibleJson = util.getJsonConfig(localConfigPath, DynamicConfigHelpersUtil.SCHEMA_TYPE_VARIABLE).get(0);
-        populatePojo(varibleJson, DynamicConfigHelpersUtil.SCHEMA_TYPE_VARIABLE);
-
-        //security
-        String securityJson = util.getJsonConfig(localConfigPath, DynamicConfigHelpersUtil.SCHEMA_TYPE_SECURITY).get(0);
-        populatePojo(securityJson, DynamicConfigHelpersUtil.SCHEMA_TYPE_SECURITY);
-
-        //table
-        populateTablesPojo(util.getJsonConfig(localConfigPath, DynamicConfigHelpersUtil.SCHEMA_TYPE_TABLE));
-    }
-
-    @SuppressWarnings("unchecked")
-    private void populatePojo(String jsonConfig, String configType) throws Exception {
-        switch (configType) {
-        case DynamicConfigHelpersUtil.SCHEMA_TYPE_VARIABLE:
-      this.variables = (Map<String, Object>) parseJsonConfig(jsonConfig,
-        DynamicConfigHelpersUtil.SCHEMA_TYPE_VARIABLE);
-      break;
+    public ElideConfigParser(String localFilePath) {
 
-        case DynamicConfigHelpersUtil.SCHEMA_TYPE_SECURITY:
-      this.elideSecurityConfig = (ElideSecurityConfig) parseJsonConfig(jsonConfig,
-        DynamicConfigHelpersUtil.SCHEMA_TYPE_SECURITY);
-      break;
+        if (DynamicConfigHelpers.isNullOrEmpty(localFilePath)) {
+            throw new IllegalArgumentException("Config path is null");
         }
-    }
-
-    private void populateTablesPojo(List<String> tablesJson) throws Exception {
-        Set<Table> tables = new HashSet<>();
-        for (String tableJson : tablesJson) {
-      ElideTableConfig table = (ElideTableConfig) parseJsonConfig(tableJson,
-              DynamicConfigHelpersUtil.SCHEMA_TYPE_TABLE);
-      tables.addAll(table.getTables());
-        }
-        this.elideTableConfig = new ElideTableConfig();
-        this.elideTableConfig.setTables(tables);
-    }
-
+        try {
+            String basePath = DynamicConfigHelpers.formatFilePath(localFilePath);
 
-    private Object parseJsonConfig(String jsonConfig, String inputConfigType) throws Exception {
-        JSONObject configType = util.schemaToJsonObject(inputConfigType);
+            this.variables = DynamicConfigHelpers.getVaribalesPojo(basePath);
+            this.elideTableConfig = DynamicConfigHelpers.getElideTablePojo(basePath);
+            this.elideSecurityConfig = DynamicConfigHelpers.getElideSecurityPojo(basePath);
 
-        if (configType != null && util.isConfigValid(configType, jsonConfig)) {
-      return util.getModelPojo(inputConfigType, jsonConfig);
-        }
-        else {
-      log.error(DynamicConfigHelpersUtil.INVALID_ERROR_MSG);
-      return null;
+        } catch (JsonProcessingException e) {
+            log.error("Error while parsing dynamic config at location " + localFilePath);
+            log.error(e.getMessage());
```

Comment:
If the config format is busted, I wonder if we should throw an IllegalStateException here.